src/rust/engine/options/src/parse.rs

```diff
@@ -274,49 +278,188 @@ fn format_parse_error(
     ))
 }
 
-#[allow(dead_code)]
-pub(crate) fn parse_bool(value: &str) -> Result<bool, ParseError> {
-    option_value_parser::bool(value).map_err(|e| format_parse_error("bool", value, e))
+pub(crate) fn parse_dict(value: &str) -> Result<DictEdit, ParseError> {
```

Comment:
We do still need an explicit `parse_dict()` though.

Reply:
Because dict values are heterogeneous, so there is no scalar type to attach their conversion to, like we do for lists. We could implement `parse()` (but not `parse_list()`) on `HashMap<String, Val>` but that doesn't really buy us anything. The point of the other cleanups was to get rid of boilerplate via type parametrization, but that doesn't apply here.